Fix configuration error in PHPUnit config file

settermjd opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

A PR build failed with the following error:

> Running before_script: xmllint --schema vendor/phpunit/phpunit/phpunit.xsd phpunit.xml.dist
> phpunit.xml.dist:7: element include: Schemas validity error : Element 'include': This element is not expected.

On reviewing [the applicable PHPUnit documentation](https://docs.phpunit.de/en/10.5/configuration.html#the-coverage-element) the `coverage` tag no longer supports an `include` tag. So, the `coverage` tag needed to be renamed to `source` as in this change.

I hope the change helps.

See:

- https://github.com/laminas/laminas-servicemanager/actions/runs/16868579885/job/47779169955
- https://docs.phpunit.de/en/10.5/configuration.html#the-coverage-element

### How to test

Run the following command pre and post-fix:

```
xmllint --schema vendor/phpunit/phpunit/phpunit.xsd phpunit.xml.dist
```

settermjd committed:
Fix configuration error in PHPUnit config file

A PR build failed with the following error:

Running before_script: xmllint --schema vendor/phpunit/phpunit/phpunit.xsd phpunit.xml.dist
phpunit.xml.dist:7: element include: Schemas validity error : Element 'include': This element is not expected.

On reviewing the applicable PHPUnit documentation the coverage tag no
longer supports an include tag. So, the coverage tag needed to be
renamed to source (in this case anyway), as in this change.

I hope it helps.

See:

- https://github.com/laminas/laminas-servicemanager/actions/runs/16868579885/job/47779169955
- https://docs.phpunit.de/en/10.5/configuration.html#the-coverage-element

Signed-off-by: Matthew Setter <matthew@matthewsetter.com>

settermjd commented:
It seems that I missed something with this PR. But, running xmllint locally succeeds, yet it fails in the CI/CD pipeline. Any input is greatly appreciated.

alexmerlin reviewed (approved):
👍

samsonasik commented:
minimum phpunit may need to be set

https://github.com/laminas/laminas-servicemanager/blob/625f2aa3bc6dd02688b2da5155b3a69870812bda/composer.json#L37

gsteel commented:
As per @samsonasik comment: `composer up && composer bump -D && composer up`

settermjd committed:
Bump package versions

As per recommendation from @samsonasik and @gsteel.

Signed-off-by: Matthew Setter <matthew@matthewsetter.com>

settermjd commented:
> As per @samsonasik comment: `composer up && composer bump -D && composer up`

Done.

settermjd commented:
Regarding the latest CI build fail, I just checked to see that static analysis isn't part of the Composer check script. Will see what I can do to resolve the issue.

settermjd commented:
Okay, Psalm found 27 issues directly but also said that it found 278 others. As that's outside the scope, as I see it, of this PR, I'm happy to address these issues, where I can, in a separate PR. Given that, can this PR be merged @gsteel @alexmerlin?

gsteel commented:
Just `psalm --set-baseline=psalm-baseline.xml` and commit the expanded baseline file.

These issues will be fixed in more recent releases anyway I'd suspect.

We should have green CI before merging 👍

settermjd committed:
Update the Psalm baseline

Doing this as per the recommendation from @gsteel:
https://github.com/laminas/laminas-servicemanager/pull/262#issuecomment-3176824964.

See
https://psalm.dev/docs/running_psalm/dealing_with_code_issues/#using-a-baseline-file
for further information.

Signed-off-by: Matthew Setter <matthew@matthewsetter.com>

settermjd commented:
Thanks for the recommendation, @gsteel regarding updating the Psalm baseline. Done.
